### PR TITLE
Making append function

### LINKED ISSUE #8: text2file override is incorrect
This override in the DM API file is incorrect:

```dm
#define text2file(text, fname) rustg_file_write(text, fname)
```

`rustg_file_write` overwrites, while `text2file` appends.

Low priority because /tg/ does not currently use the `file` module or the `RUSTG_OVERRIDE_BUILTINS` feature.

### PR DESCRIPTION
**Pull request by Sindorman**
This PR adds append function for the file. Fixes #8

### CODE REVIEW

**Comment by SpaceManiac** [comment]
I think I would prefer if we added a new function `file_append` and changed the `text2file` override to use that. There's some use in keeping the overwriting version around (namely, it can replace any `fdel`-`text2file` sequence). Thoughts?

**Comment by Sindorman** [comment]
I have got no problem with it. @SpaceManiac . Changed.

**Comment by SpaceManiac** [comment]
Looks good so far. The new function needs an entry in `build.rs` under the `// module: file` section, and the `text2file` override should use that new function instead.

File: src/file.rs [hunk 1]
```diff
@@ -26,3 +31,9 @@ fn write(data: &str, path: &str) -> Result<usize> {
 
     Ok(file.write(data.as_bytes())?)
 }
+
+fn append(data: &str, path: &str) -> Result<usize> {
+    let file = OpenOptions::new().append(true).create(true).open(path)?;
```
Comment (SpaceManiac) [inline src/file.rs]: Needs to be mutable for the `write()` call to work:
```suggestion
    let mut file = OpenOptions::new().append(true).create(true).open(path)?;
```
I'll tweak Travis to catch this.

**Pull request merged**